packages/aws-cdk/lib/settings.ts

```diff
@@ -135,6 +140,7 @@ export class Configuration {
     this.settings = this.defaultConfig
       .merge(userConfig)
       .merge(this.projectConfig)
+      .merge(localConfig)
```

Comment:
I think it makes sense to override `cdk.json` but still let CLI take precedence.